platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/signed/SignedState.java

```diff
@@ -702,6 +732,41 @@ public void pruneInvalidSignatures(@NonNull final AddressBook trustedAddressBook
         }
     }
 
+    /**
+     * Remove all invalid signatures from a signed state.
+     *
+     * @param trustedRoster use this roster to determine signature validity instead of using the roster from the signed
+     *                      state. (Useful if validating signed states from untrusted sources.)
+     */
+    public void pruneInvalidSignatures(@NonNull final Roster trustedRoster) {
+        Objects.requireNonNull(trustedRoster);
+
+        final Map<Long, RosterEntry> entriesByNodeId = RosterUtils.toMap(trustedRoster);
+        final List<NodeId> signaturesToRemove = new ArrayList<>();
+
+        for (final NodeId nodeId : sigSet) {
+            final RosterEntry entry = entriesByNodeId.get(nodeId.id());
+            if (!isSignatureValid(entry, sigSet.getSignature(nodeId))) {
+                signaturesToRemove.add(nodeId);
+            }
+        }
+
+        for (final NodeId nodeId : signaturesToRemove) {
+            sigSet.removeSignature(nodeId);
+        }
+
+        long newWeight = 0;
```

Comment:
The comment from the equivalent method should also be copied so it is not totally gone when the other method is cleaned up